source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesScriptServiceV1IntegrationTest.cs

```diff
@@ -59,4 +59,66 @@ Task ScriptCompleted(CancellationToken ct)
             return Task.CompletedTask;
         }
     }
+
+    [Test]
+    public async Task TentaclePodIsTerminatedDuringScriptExecution_ShouldRestartAndPickUpPodStatus()
+    {
+        // Arrange
+        var logs = new List<ProcessOutput>();
+        var scriptCompleted = false;
+        var count = 50;
+        var scriptBody = $@"echo ""Hello World""
+for i in $(seq 1 {50});
+do
+    echo Count: $i
+    sleep 0.1
+done
+";
+        var command = new ExecuteKubernetesScriptCommandBuilder(LoggingUtils.CurrentTestHash())
+            .WithScriptBody(scriptBody)
+            .Build();
+
+        var commandResult = await KubeCtl.ExecuteNamespacedCommand("get pods -l app.kubernetes.io/name=octopus-agent -o \"Name\"");
+        var initialPodName = commandResult.StdOut.Single();
+
+        //act
+        var scriptTask = TentacleClient.ExecuteScript(command, StatusReceived, ScriptCompleted, new InMemoryLog(), CancellationToken.None);
+        var killTask = KubeCtl.ExecuteNamespacedCommand("delete pods -l app.kubernetes.io/name=octopus-agent");
+
+        await Task.WhenAll(scriptTask, killTask);
+
+        var result = scriptTask.Result;
+
+        commandResult = await KubeCtl.ExecuteNamespacedCommand("get pods -l app.kubernetes.io/name=octopus-agent -o \"Name\"");
+        var finalPodName = commandResult.StdOut.Single();
+
+        //Assert
+        logs.Should().Contain(po => po.Source == ProcessOutputSource.StdOut && po.Text == "Hello World");
```

Comment:
Is it possible to also assert that Halibut got an error and is retrying mid way through the script?

Reply:
🤔 Possibly yes

Reply:
I actually don't think this test is working correctly. I may need to tweak the script so that it waits until the old pod is dead (or something like that)